ada_feeding/ada_feeding/behaviors/moveit2/modify_collision_object.py

```diff
@@ -0,0 +1,230 @@
+            if collision_object_orientation is None:
+                raise ValueError(
+                    "The collision object orientation input key must be specified for "
+                    "ADD and MOVE operations."
+                )
+        if operation == ModifyCollisionObjectOperation.ADD:
+            mesh_filepath = self.blackboard_get("mesh_filepath")
+            prim_type = self.blackboard_get("prim_type")
+            dims = self.blackboard_get("dims")
+            if mesh_filepath is None and (prim_type is None or dims is None):
+                raise ValueError(
+                    "If `mesh_filepath` is None, then both `prim_type` and `dims` "
+                    "must be specified."
+                )
+
+    @override
+    def update(self) -> py_trees.common.Status:
+        # Docstring copied from @override
+
+        self.logger.info(f"{self.name} [ModifyCollisionObject::update()]")
+
+        # Get the blackboard inputs for all operations
+        operation = self.blackboard_get("operation")
+        collision_object_id = self.blackboard_get("collision_object_id")
+
+        # Remove the collision object
+        if operation == ModifyCollisionObjectOperation.REMOVE:
+            with self.moveit2_lock:
+                self.moveit2.remove_collision(collision_object_id)
+            return py_trees.common.Status.SUCCESS
+
+        # Get the blackboard inputs for ADD and MOVE operations
+        collision_object_position = self.blackboard_get("collision_object_position")
+        collision_object_orientation = self.blackboard_get(
+            "collision_object_orientation"
+        )
+        frame_id = self.blackboard_get("frame_id")
+        mesh_filepath = self.blackboard_get("mesh_filepath")
+        prim_type = self.blackboard_get("prim_type")
+        dims = self.blackboard_get("dims")
+        position_offset = self.blackboard_get("position_offset")
+
+        # Update types
+        if isinstance(collision_object_position, PointStamped):
+            frame_id = collision_object_position.header.frame_id
+            collision_object_position = Point(
+                x=collision_object_position.point.x + position_offset[0],
+                y=collision_object_position.point.y + position_offset[1],
+                z=collision_object_position.point.z + position_offset[2],
+            )
+        elif isinstance(collision_object_position, tuple):
+            collision_object_position = Point(
+                x=collision_object_position[0] + position_offset[0],
+                y=collision_object_position[1] + position_offset[1],
+                z=collision_object_position[2] + position_offset[2],
+            )
+        if isinstance(collision_object_orientation, QuaternionStamped):
+            frame_id = collision_object_orientation.header.frame_id
+            collision_object_orientation = collision_object_orientation.quaternion
+
+        # Move the collision object
+        with self.moveit2_lock:
```

Comment:
Always check is_locked before a with statement so we don't block the tree thread.